### PR TITLE
MHV-48175 Handle Vaccines API errors

### PR DESCRIPTION
**Pull request by tynyttie**
## Summary
Frontend error handling for API errors on the Vaccines page has been integrated.
## Related issue(s)

[MHV-48175](https://jira.devops.va.gov/browse/MHV-48175) - Handle Vaccines API errors

Handle Vaccines API errors

User Story: As a user, I want to see an appropriate error message when API calls fail, so that I can retry or take other appropriate steps.  

Acceptance Criteria:
AC1 All API calls for this domain handle errors gracefully
AC2
AC3
AC4
AC5

## Testing done
The appearance of an error message was triggered by issues with the vaccines. A unit test was subsequently developed to evaluate the code implementation.
Screenshot 2023-10-17 at 11 22 37 AM (2)
Screenshot 2023-10-17 at 11 23 01 AM (2)

## What areas of the site does it impact?
The modification affects the Vaccine section of the Medical Records page.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by mattwrightva — COMMENTED** [review]
Hey Ty, check out tests/actions/allergies.unit.spec.js for examples of how to write unit tests to cover the lines you added here. Compare with tests/actions/vaccines.unit.spec.js and you'll see what needs to be added.

**Review by mmoyer-va — CHANGES_REQUESTED** [review]
In addition to dispatching an alert when something goes wrong, we will need to display that alert in the Vaccines list and details pages. We are using `AccessTroubleAlertBox` to print the alert, but it is currently specific to allergies. It would need modifications if we want to use it for other domains.

File: src/applications/mhv/medical-records/api/MrApi.js [hunk 1]
```diff
@@ -145,19 +143,9 @@ export const getAllergy = id => {
...
-  }
-  return new Promise(resolve => {
-    setTimeout(() => {
-      resolve(vaccines);
-    }, 1000);
+export const getVaccineList = () => {
+  return apiRequest(`${apiBasePath}/medical_records/vaccines`, {
+    headers,
   });
 };
```
Comment (mmoyer-va) [inline src/applications/mhv/medical-records/api/MrApi.js:150]: I am in favor making the Vaccine APIs use real data, but were you and Matt able to access vaccine data from the backend? I just tried it locally and it prints your error. I want to make sure it works before we push this change to staging.

I looked into it and the error I'm seeing is in the vaccines reducer. It's not checking if the response is empty before trying to grab records.
```
ERROR! TypeError: Cannot read properties of undefined (reading 'map')
    at vaccineReducer (vaccines.js:58:12)
```

**Review by mmoyer-va — CHANGES_REQUESTED** [review]
A few changes:
- The alert box for Vaccines says "Allergies". That component will need a type prop or something.
- There is an error in the vaccines reducer (see comment)
- Please include a screenshot with your PR, as this involves frontend changes.
- Your PR should quote the Jira ticket, per our recent guidelines

**Review by mmoyer3 — APPROVED** [review]
Looks good!

**Review by mattwrightva — APPROVED** [review]
Awesome!

**Review by mmoyer-va — APPROVED** [review]

**Pull request merged**